### PR TITLE
Client Encryption: Adds Microsoft.Azure.Cosmos compatibility to version 3.35.4

### LINKED ISSUE #4207: Microsoft.Azure.Cosmos.Encryption has dependency on Microsoft.Azure.Cosmos with max version 3.35.2 while min recommended versions is 3.35.4
I'm unable to upgrade nuget packages which is following by recommendation and upgraded their min version Microsoft.Azure.Cosmos to version 3.35.4. 

Is it possible to fix this?

### PR DESCRIPTION
**Pull request by vipulvishal-ms**
## Description

Compatibility for Microsoft.Azure.Cosmos changed to version 3.35.4.

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #4207

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj [hunk 1]
```diff
@@ -28,11 +28,11 @@
   </ItemGroup>
 	
   <ItemGroup Condition=" '$(SdkProjectRef)' != 'True' AND '$(IsPreview)' != 'True' ">
-    <PackageReference Include="Microsoft.Azure.Cosmos" Version="[3.31.0,3.35.2]" />
+    <PackageReference Include="Microsoft.Azure.Cosmos" Version="[3.31.0,3.35.4]" />
```
Comment (kr-santosh) [inline Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:31]: @vipulvishal-ms  Just check if we can support the lastest version ( [3.37.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.37.0) of Azure.Cosmos and set it to that. 

@ealsur @kirankumarkolli do we set the minimum required to 3.35.4? Is there a reason why we are recommending it?
https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/sdk-dotnet-v3#recommended-version

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj [hunk 2]
```diff
@@ -28,11 +28,11 @@
...
   </ItemGroup>
 
   <ItemGroup Condition=" '$(SdkProjectRef)' != 'True' AND '$(IsPreview)' == 'True' ">
-    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.35.2-preview" />
+    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.35.4-preview" />
```
Comment (kr-santosh) [inline Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:35]: same just check if we can support the latest

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj (hunk 1 above)
Comment (vipulvishal-ms) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: yes, 3.37.0 is compatible. Changed it to 3.37.0.

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Recommended version is used to mark the version that has the **critical** bug fixes.

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Customers should use **at least** that version. They can go higher if they want, for new features for example

Comment (vipulvishal-ms) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: @ealsur so can we make 3.35.4 as min supported version ?

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Makes sense

**Review by kr-santosh — DISMISSED** [review]

File: Directory.Build.props [hunk 3]
```diff
@@ -4,8 +4,8 @@
 		<ClientPreviewVersion>3.37.0</ClientPreviewVersion>
 		<ClientPreviewSuffixVersion>preview</ClientPreviewSuffixVersion>
 		<DirectVersion>3.31.5</DirectVersion>
-		<EncryptionOfficialVersion>2.0.3</EncryptionOfficialVersion>
```
Comment (ealsur) [inline Directory.Build.props:7]: @kirankumarkolli / @Pilchie would this be a good time to enforce our new [versioning](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/docs/versioning.md) schema, even if this is the Encryption package?

If so, this should be 2.0.3 and 2.1.0-preview3.

Comment (Pilchie) [reply Directory.Build.props:7]: Yes, I agree that we should adpot the new versioning scheme here.

Comment (kr-santosh) [reply Directory.Build.props:7]: so, the previews have a higher minor version representation w.r.t to stable version? @ealsur

**Review by ealsur — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**